stdlib/ballerina-http/src/main/ballerina/http/Package.md

```diff
@@ -36,6 +36,43 @@ See [Mutual SSL Example](https://ballerinalang.org/docs/by-example/mutual-ssl).
 
 See [Caching Example](https://ballerinalang.org/docs/by-example/caching), [HTTP Disable Chunking Example](https://ballerinalang.org/docs/by-example/http-disable-chunking).
 
+###WebSockets
+The package also provides support for WebSockets. There are two types of WebSocket endpoint which are `WebSocketClient` and `WebSocketListener`. Both endpoints support all WebSocket frames. The `WebSocketClient` has a callback service.
+
+There are also two types of services in WebSocket which are `WebSocketService` and `WebSocketClientService`. The callback service for `WebSocketClient` would always be a `WebSocketClientService`. The WebSocket services have a fixed set of resources which do not have a resource config. The incoming messages are passed to these resources.
+
+**WebSocket upgrade**: During a WebSocket upgrade the initial message would be an HTTP request. If this request is to be intercepted and the handshake be made explicitly by the user with custom headers the user has to create a HTTP resource with WebSocket specific configurations as follows:
```

Comment:
Rewrite this paragraph as: "During a WebSocket upgrade, the initial message is an HTTP request. To intercept this request and make the handshake explicitly with custom headers, the user must create an HTTP resource with WebSocket specific configurations as follows:"

Reply:
Please verify that I'm understanding the meaning correctly. It's the custom headers that intercept the message and make the handshake?